app/javascript/mastodon/models/poll.ts

```diff
@@ -31,79 +21,62 @@ export function createPollOptionTranslationFromServerJSON(
 ) {
-  return PollOptionTranslationFactory({
+  return {
     ...translation,
     titleHtml: emojify(
       escapeTextContentForBrowser(translation.title),
       emojiMap,
     ),
-  });
+  } as PollOptionTranslation;
 }
 
-export type PollOption = RecordOf<PollOptionShape>;
-
-export const PollOptionFactory = Record<PollOptionShape>({
-  title: '',
-  votes_count: 0,
-  voted: false,
-  titleHtml: '',
-  translation: null,
-});
-
-interface PollShape
+export interface Poll
   extends Omit<ApiPollJSON, 'emojis' | 'options' | 'own_votes'> {
-  emojis: List<CustomEmoji>;
-  options: List<PollOption>;
-  own_votes?: List<number>;
+  emojis: CustomEmoji[];
+  options: PollOption[];
+  own_votes?: number[];
 }
-export type Poll = RecordOf<PollShape>;
 
-export const PollFactory = Record<PollShape>({
-  id: '',
-  expires_at: '',
+const pollDefaultValues = {
   expired: false,
   multiple: false,
   voters_count: 0,
   votes_count: 0,
   voted: false,
-  emojis: List<CustomEmoji>(),
-  options: List<PollOption>(),
-  own_votes: List(),
-});
+  own_votes: [],
+};
 
 export function createPollFromServerJSON(
   serverJSON: ApiPollJSON,
   previousPoll?: Poll,
 ) {
   const emojiMap = makeEmojiMap(serverJSON.emojis);
 
-  return PollFactory({
+  return {
+    ...pollDefaultValues,
     ...serverJSON,
-    emojis: List(serverJSON.emojis.map((emoji) => CustomEmojiFactory(emoji))),
-    own_votes: serverJSON.own_votes ? List(serverJSON.own_votes) : undefined,
-    options: List(
-      serverJSON.options.map((optionJSON, index) => {
-        const option = PollOptionFactory({
-          ...optionJSON,
-          voted: serverJSON.own_votes?.includes(index) || false,
-          titleHtml: emojify(
-            escapeTextContentForBrowser(optionJSON.title),
-            emojiMap,
-          ),
-        });
-
-        const prevOption = previousPoll?.options.get(index);
-        if (prevOption?.translation && prevOption.title === option.title) {
-          const { translation } = prevOption;
-
-          option.set(
-            'translation',
-            createPollOptionTranslationFromServerJSON(translation, emojiMap),
-          );
-        }
-
-        return option;
-      }),
-    ),
-  });
+    emojis: serverJSON.emojis.map((emoji) => CustomEmojiFactory(emoji)),
+    options: serverJSON.options.map((optionJSON, index) => {
+      const option = {
+        ...optionJSON,
+        voted: serverJSON.own_votes?.includes(index) || false,
+        titleHtml: emojify(
+          escapeTextContentForBrowser(optionJSON.title),
+          emojiMap,
+        ),
+      } as PollOption;
+
+      const prevOption = previousPoll?.options[index];
+      if (prevOption?.translation && prevOption.title === option.title) {
+        const { translation } = prevOption;
+
+        option.translation = createPollOptionTranslationFromServerJSON(
+          translation,
+          emojiMap,
+        );
+      }
+
+      return option;
+    }),
+  } as Poll;
```

Comment:
Is it me, or is `own_votes` dropped here?

Reply:
`own_votes` is a plain array, so it is now file to keep it as it was in the original JSON object, while before it needed to be converted to an Immutable list.